### PR TITLE
Use @rnx-kit/cli for metro bundling and serving

### PR DESCRIPTION
**Pull request by afoxman**
### Platforms Impacted
- [x] iOS
- [x] macOS
- [x] win32 (Office)
- [x] windows
- [x] android

### Description of changes

Remove metro logic from FURN, replacing it with calls to the @rnx-kit/cli package (via @react-native-community/cli). Use rnx-bundle for bundling, and rnx-start for running a bundle server.

package.json: 
* update scripts to use rnx-bundle / rnx-start
* convert bundling config from metroBundles to rnx-kit config

justPreset.ts:
* remove metro build commands and options
 
### Verification

Ran local builds and bundle commands.
Ran fluentui test app for ios and macos.

### Pull request checklist

This PR has considered (when applicable):
- [x] Automated Tests
- [x] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

**Comment by Saadnajmi** [comment]
Since I couldn't find it online, what is rnx-kit?

File: apps/win32/package.json [hunk 1]
```diff
@@ -15,10 +15,10 @@
     "code-style": "fluentui-scripts code-style",
     "depcheck": "fluentui-scripts depcheck",
     "lint": "fluentui-scripts eslint",
-    "start": "fluentui-scripts metro --server --cli --port 8081",
+    "start": "react-native rnx-bundle --server --port 8081",
```
Comment (tido64) [inline apps/win32/package.json:18]: Shouldn't this be `rnx-start`?
```suggestion
    "start": "react-native rnx-start",
```

Comment (afoxman) [reply apps/win32/package.json]: Yes. :)

**Comment by afoxman** [comment]
@Saadnajmi Sorry - I should have explained. @rnx-kit is an expansion of the work in sdx-platform and fabric-internal. It is intended to be a Microsoft-wide react-native SDK. A few of us have just recently started the work. In this case, I built a CLI plugin to react-native-community/cli which provides metro bundling. This eliminates the need for custom metro code in repos building react-native apps. The github repo isn't public yet, which is why you aren't seeing it.

**Review by tido64 — APPROVED** [review]

**Comment by afoxman** [comment]
@JasonVMo can you review/approve? For some reason, tido64's approval isn't letting me merge.

**Review by JasonVMo — APPROVED** [review]
The changes look good to me.

**Pull request merged**